Games/Retro_ML.SuperMario64/Neural/Scoring/DiedScoreFactor.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Retro_ML.Configuration.FieldInformation;
+using Retro_ML.Game;
+using Retro_ML.Neural.Scoring;
+using Retro_ML.SuperMario64.Game;
+using Retro_ML.SuperMario64.Game.Data;
+
+namespace Retro_ML.SuperMario64.Neural.Scoring;
+internal class DiedScoreFactor : IScoreFactor
+{
+    private bool shouldStop;
+    private double currScore;
+
+    public FieldInfo[] Fields => Array.Empty<FieldInfo>();
+
+    public DiedScoreFactor() => ExtraFields = Array.Empty<ExtraField>();
+
+    public object this[string fieldName] { get => 0; set { } }
+    public bool ShouldStop => shouldStop;
+    public double ScoreMultiplier { get; set; }
+
+    public string Name => "Death";
+    public string Tooltip => "Reward applied on AI death.";
+
+    public bool CanBeDisabled => false;
```

Comment:
pas mieux de laisser la possibilité de disable pour le playMode?

Reply:
Since Mario gets kicked out of the level when he dies, the simulation will need to be stopped